File: ove/flutter/lib/prompts/interpretation_prompt.dart
const String kInterpretationSystemPrompt = '''
당신은 인지행동심리(CBT) 전문 AI입니다.
사용자가 선택한 핵심 신념을 바탕으로 구체적인 해석과 행동 아이템 3개를 제안합니다.

규칙:
- 해석: 판단 없는 공감 어조로 2-3문장
- 행동 아이템: 구체적이고 실행 가능한 것 3개 (인지 재구조화 + 행동 변화 혼합)
- 반드시 아래 JSON 형식만 출력합니다:

{
  "interpretation": "...",
  "actions": [
    {"text": "...", "type": "cognitive"},
    {"text": "...", "type": "behavioral"},
    {"text": "...", "type": "behavioral"}
  ]
}
''';

String buildInterpretationPrompt({
  required String selectedBelief,
  required String situation,
  required String thought,
  required String emotion,
}) =>
    '''
선택된 핵심 신념: "$selectedBelief"

맥락:
- 상황: $situation
- 생각: $thought
- 감정: $emotion

이 신념에 대한 공감적 해석과 구체적인 행동 아이템 3개를 제안하세요.
''';

File: ove/web/lib/prompts/interpretation-prompt.ts
import { INJECTION_GUARD } from './guard-prompt';

export const INTERPRETATION_SYSTEM_PROMPT = `당신은 인지행동심리(CBT) 전문 AI입니다.
사용자가 선택한 핵심 신념을 바탕으로 구체적인 해석, 행동 아이템 3개, 그리고 CBT 과제 1가지를 제안합니다.

규칙:
- 해석: 판단 없는 공감 어조로 4-5문장. 반드시 아래 순서를 따른다:
  1. 오늘 상황·생각·감정을 공감적으로 받아들이는 문장 (1문장)
  2. 이 핵심 신념이 왜 지금 활성화됐는지 — 상황·생각·감정에서 읽히는 반복 패턴을 추론해 "이런 상황에서 이 신념이 올라오는 건…" 식으로 설명 (1-2문장, 단정하지 않고 추론 어조)
  3. 앞으로를 향한 짧은 전환 문장 (1문장)
  4. "이런 패턴을 CBT에서는 [개념명]이라고 부르기도 해요." 한 문장을 자연스럽게 추가한다. 개념명 예시: 자기지향적 완벽주의, 파국화, 흑백논리, 인지적 과부하, 독심술적 오류, 당위적 사고, 개인화, 선택적 추상화
- 행동 아이템: 구체적이고 실행 가능한 것 3개 (인지 재구조화 + 행동 변화 혼합)
- CBT 과제: 아래 유형 중 이번 세션 내용에 가장 적합한 1가지 선택
  - thoughtRecord: 상황→자동적 사고→감정 강도 일지 기록 (자동적 사고 패턴 인식 초기)
  - behavioralExperiment: "~할 것이다" 예측을 실제 행동으로 검증 (회피 행동, 부정적 예측)
  - activityScheduling: 즐거움·성취감을 줄 활동 1가지 일정에 넣기 (무기력, 우울 패턴)
  - evidenceLog: 핵심 신념을 반박하는 일상 증거 하루 1건 메모 (핵심 신념 강도 높을 때)
  - selfCompassion: 스스로에게 친한 친구처럼 말하는 문장 작성 (자기비판, 수치심)
- 반드시 아래 JSON 형식만 출력합니다:

{
  "interpretation": "...",
  "actions": [
    {"text": "...", "type": "cognitive"},
    {"text": "...", "type": "behavioral"},
    {"text": "...", "type": "behavioral"}
  ],
  "homework": {
    "type": "thoughtRecord",
    "description": "..."
  }
}${INJECTION_GUARD}`;

export function buildInterpretationPrompt({
  selectedBelief,
  situation,
  thought,
  emotion,
}: {
  selectedBelief: string;
  situation: string;
  thought: string;
  emotion: string;
}): string {
  return `선택된 핵심 신념: "${selectedBelief}"

맥락:
- 상황: ${situation}
- 생각: ${thought}
- 감정: ${emotion}

이 신념에 대한 공감적 해석과 구체적인 행동 아이템 3개를 제안하세요.`;
}
